### PR TITLE
Fix memory leak

### PR DESCRIPTION
**Pull request by rgachuhi**
This PR address a memory leak issue in svn sync manager

### CODE REVIEW

File: src/main/java/edu/cmu/oli/content/svnmanager/SVNManager.java [hunk 1]
```diff
@@ -330,7 +326,11 @@ public boolean isVersioned(File path) {
     }
 
     public List<File> listModifiedFiles(File path) throws SVNException {
-        SVNClientManager svnClientManager = SVNClientManager.newInstance();
+        SVNClientManager svnClientManager = clientManagerMap.get(path.getAbsolutePath());
+        if(svnClientManager == null) {
```
Comment (darrensiegel) [inline src/main/java/edu/cmu/oli/content/svnmanager/SVNManager.java:330]: These `svnClientManager` accesses should be guarded with a `synchronized` block in order to guarantee cross-thread visibility. Without synchronization the following is possible:

1. Thread A gets a `null` on line 330 and thus creates and inserts into the HashMap a new `SVNClientManager`.
2. Thread B comes along after A, and attempts to access the same `SVNClientManager` (aka from the same absolute path) but also receives a `null`. Thread B recreates the `SVNClientManager` and inserts it into the HashMap, probably leaving the one that A created 'leaked'.

**Review by darrensiegel — CHANGES_REQUESTED** [review]
See note inline about missing synchronization

File: src/main/java/edu/cmu/oli/content/svnmanager/SVNManager.java (hunk 1 above)
Comment (rgachuhi) [reply src/main/java/edu/cmu/oli/content/svnmanager/SVNManager.java:331]: synchronized block added

File: src/main/java/edu/cmu/oli/content/svnmanager/SVNManager.java [hunk 2]
```diff
@@ -342,8 +343,12 @@ public boolean isVersioned(File path) {
         return fileList;
     }
 
-    public List<File> listAddedFiles(File path) throws SVNException {
-        SVNClientManager svnClientManager = SVNClientManager.newInstance();
+    public synchronized List<File> listAddedFiles(File path) throws SVNException {
+        SVNClientManager svnClientManager = clientManagerMap.get(path.getAbsolutePath());
+        if(svnClientManager == null) {
```
Comment (darrensiegel) [inline src/main/java/edu/cmu/oli/content/svnmanager/SVNManager.java:348]: The use of `ConcurrentHashMap` here alone doesn't make this implementation "correct" with regards to thread safety.  With a context switch at just the right time, two different threads will see the `svnClientManager` as `null` and then create and put a new instance. A thread safe solution here is something like: 

```
SVNClientManager svnClientManager;
synchronized (clientManagerMap) {
  svnClientManager = clientManagerMap.get(path.getAbsolutePath());
  if (svnClientManager == null) {
    svnClientManager = SVNClientManager.newInstance();
    clientManagerMap.put(path.getAbsolutePath(), svnClientManager);
  }
}
```

**Review by darrensiegel — CHANGES_REQUESTED** [review]

File: src/main/java/edu/cmu/oli/content/svnmanager/SVNManager.java (hunk 2 above)
Comment (rgachuhi) [reply src/main/java/edu/cmu/oli/content/svnmanager/SVNManager.java]: Ah. Makes sense. Updated with suggested changes.

**Review by darrensiegel — APPROVED** [review]

**Pull request merged**